demos/coreMQTT/mqtt_demo_mutual_auth.c

```diff
@@ -0,0 +1,981 @@
+                    LogError( ( "MQTT_ProcessLoop failed: LoopDuration=%u, Error=%s\r\n",
+                                mqttexamplePROCESS_LOOP_TIMEOUT_MS,
+                                MQTT_Status_strerror( xMQTTStatus ) ) );
+                }
+            }
+
+            /* Leave Connection Idle for some time. */
+            LogInfo( ( "Keeping Connection Idle...\r\n\r\n" ) );
+            vTaskDelay( mqttexampleDELAY_BETWEEN_PUBLISHES_TICKS );
+        }
+
+        /************************ Unsubscribe from the topic. **************************/
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            LogInfo( ( "Unsubscribe from the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+            lReturnStatus = prvMQTTUnsubscribeFromTopic( &xMQTTContext );
+        }
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            /* Process incoming UNSUBACK packet from the broker. */
+            xMQTTStatus = MQTT_ProcessLoop( &xMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                lReturnStatus = EXIT_FAILURE;
+                LogError( ( "Failed to receive UNSUBACK packet from broker: ProcessLoopDuration=%u, Error=%s\r\n",
+                            mqttexamplePROCESS_LOOP_TIMEOUT_MS,
+                            MQTT_Status_strerror( xMQTTStatus ) ) );
+            }
+        }
+
+        /**************************** Disconnect. ******************************/
+
+        /* Send an MQTT Disconnect packet over the already connected TLS over TCP connection.
+         * There is no corresponding response for the disconnect packet. After sending
+         * disconnect, client must close the network connection. */
+        LogInfo( ( "Disconnecting the MQTT connection with %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+        MQTT_Disconnect( &xMQTTContext );
+
+        /* Close the network connection.  */
+        SecureSocketsTransport_Disconnect( &xNetworkContext );
+
+        /* Reset SUBACK status for each topic filter after completion of subscription request cycle. */
+        for( ulTopicCount = 0; ulTopicCount < mqttexampleTOPIC_COUNT; ulTopicCount++ )
+        {
+            xTopicFilterContext[ ulTopicCount ].xSubAckStatus = MQTTSubAckFailure;
+        }
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            /* Wait for some time between two iterations to ensure that we do not
+             * the broker. */
+            LogInfo( ( "prvMQTTDemoTask() completed an iteration successfully. Total free heap is %u.\r\n", xPortGetFreeHeapSize() ) );
+            LogInfo( ( "Demo iteration %lu completed successfully.\r\n", ( ulDemoRunCount + 1UL ) ) );
+        }
+        else
+        {
+            /* code */
+            LogInfo( ( "Demo iteration %lu failed.\r\n", ( ulDemoRunCount + 1UL ) ) );
+
+            /* Reset the return value for the next iteration. */
+            lReturnStatus = EXIT_SUCCESS;
```

Comment:
So if one iteration failed we overwrite it and return success?

Reply:
I will update this logic to break out of the loop

Reply:
I updated the plaintext demo to break only at the end of the loop. Please take a look as well.